File: Runtime/OutputFlow.vb
'*********************************************************************************************
' STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************

Class OutputFlow

    Private m_FromStratumId As Integer
    Private m_FromSecondaryStratumId As Nullable(Of Integer)
    Private m_FromTertiaryStratumID As Nullable(Of Integer)
    Private m_FromStateClassId As Integer
    Private m_FromStockTypeId As Integer
    Private m_transitionTypeId As Nullable(Of Integer)
    Private m_ToStratumId As Integer
    Private m_ToStateClassId As Integer
    Private m_ToStockTypeId As Integer
    Private m_FlowTypeId As Integer
    Private m_Amount As Double

    Public Sub New(
        ByVal fromStratumId As Integer,
        ByVal fromSecondaryStratumId As Nullable(Of Integer),
        ByVal fromTertiaryStratumId As Nullable(Of Integer),
        ByVal fromStateClassId As Integer,
        ByVal fromStockTypeId As Integer,
        ByVal transitionTypeId As Nullable(Of Integer),
        ByVal toStratumId As Integer,
        ByVal toStateClassId As Integer,
        ByVal toStockTypeId As Integer,
        ByVal flowTypeId As Integer,
        ByVal amount As Double)

        Me.m_FromStratumId = fromStratumId
        Me.m_FromSecondaryStratumId = fromSecondaryStratumId
        Me.m_FromTertiaryStratumID = fromTertiaryStratumId
        Me.m_FromStateClassId = fromStateClassId
        Me.m_FromStockTypeId = fromStockTypeId
        Me.m_transitionTypeId = transitionTypeId
        Me.m_ToStratumId = toStratumId
        Me.m_ToStateClassId = toStateClassId
        Me.m_ToStockTypeId = toStockTypeId
        Me.m_FlowTypeId = flowTypeId
        Me.m_Amount = amount

    End Sub

    Public ReadOnly Property FromStratumId As Integer
        Get
            Return Me.m_FromStratumId
        End Get
    End Property

    Public ReadOnly Property FromSecondaryStratumId As Nullable(Of Integer)
        Get
            Return Me.m_FromSecondaryStratumId
        End Get
    End Property

    Public ReadOnly Property FromTertiaryStratumId As Nullable(Of Integer)
        Get
            Return Me.m_FromTertiaryStratumID
        End Get
    End Property

    Public ReadOnly Property FromStateClassId As Integer
        Get
            Return Me.m_FromStateClassId
        End Get
    End Property

    Public ReadOnly Property FromStockTypeId As Integer
        Get
            Return Me.m_FromStockTypeId
        End Get
    End Property

    Public ReadOnly Property TransitionTypeId As Nullable(Of Integer)
        Get
            Return Me.m_transitionTypeId
        End Get
    End Property

    Public ReadOnly Property ToStratumId As Integer
        Get
            Return Me.m_ToStratumId
        End Get
    End Property

    Public ReadOnly Property ToStateClassId As Integer
        Get
            Return Me.m_ToStateClassId
        End Get
    End Property

    Public ReadOnly Property ToStockTypeId As Integer
        Get
            Return Me.m_ToStockTypeId
        End Get
    End Property

    Public ReadOnly Property FlowTypeId As Integer
        Get
            Return Me.m_FlowTypeId
        End Get
    End Property

    Public Property Amount As Double
        Get
            Return Me.m_Amount
        End Get
        Set(ByVal value As Double)
            Me.m_Amount = value
        End Set
    End Property

End Class

File: Runtime/OutputFlow.cs
//*********************************************************************************************
// STSimStockFlow: A SyncroSim Module for the ST-Sim Stocks and Flows Add-In.
//
// Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
//
//*********************************************************************************************

namespace SyncroSim.STSimStockFlow
{
	internal class OutputFlow
	{
		private int m_FromStratumId;
		private int? m_FromSecondaryStratumId;
		private int? m_FromTertiaryStratumID;
		private int m_FromStateClassId;
		private int m_FromStockTypeId;
		private int? m_transitionTypeId;
		private int m_ToStratumId;
		private int m_ToStateClassId;
		private int m_ToStockTypeId;
		private int m_FlowTypeId;
		private double m_Amount;

		public OutputFlow(
            int fromStratumId, int? fromSecondaryStratumId, int? fromTertiaryStratumId, int fromStateClassId, int fromStockTypeId, 
            int? transitionTypeId, int toStratumId, int toStateClassId, int toStockTypeId, int flowTypeId, double amount)
		{
			this.m_FromStratumId = fromStratumId;
			this.m_FromSecondaryStratumId = fromSecondaryStratumId;
			this.m_FromTertiaryStratumID = fromTertiaryStratumId;
			this.m_FromStateClassId = fromStateClassId;
			this.m_FromStockTypeId = fromStockTypeId;
			this.m_transitionTypeId = transitionTypeId;
			this.m_ToStratumId = toStratumId;
			this.m_ToStateClassId = toStateClassId;
			this.m_ToStockTypeId = toStockTypeId;
			this.m_FlowTypeId = flowTypeId;
			this.m_Amount = amount;
		}

		public int FromStratumId
		{
			get
			{
				return this.m_FromStratumId;
			}
		}

		public int? FromSecondaryStratumId
		{
			get
			{
				return this.m_FromSecondaryStratumId;
			}
		}

		public int? FromTertiaryStratumId
		{
			get
			{
				return this.m_FromTertiaryStratumID;
			}
		}

		public int FromStateClassId
		{
			get
			{
				return this.m_FromStateClassId;
			}
		}

		public int FromStockTypeId
		{
			get
			{
				return this.m_FromStockTypeId;
			}
		}

		public int? TransitionTypeId
		{
			get
			{
				return this.m_transitionTypeId;
			}
		}

		public int ToStratumId
		{
			get
			{
				return this.m_ToStratumId;
			}
		}

		public int ToStateClassId
		{
			get
			{
				return this.m_ToStateClassId;
			}
		}

		public int ToStockTypeId
		{
			get
			{
				return this.m_ToStockTypeId;
			}
		}

		public int FlowTypeId
		{
			get
			{
				return this.m_FlowTypeId;
			}
		}

		public double Amount
		{
			get
			{
				return this.m_Amount;
			}
			set
			{
				this.m_Amount = value;
			}
		}
	}
}